src/world/components/lib/checkTransitionMatch.js

```diff
@@ -0,0 +1,14 @@
+const _ = require('lodash');
+const Type = require('../../../components/type.js');
+
+module.exports = function checkTransitionMatch(rule, thing, locations, action) {
+  const ruleSource = rule.getSource();
+  const targetLocation = rule.getConsequentTarget();
+  const sourceMatch = ruleSource instanceof Type
```

Comment:
`ruleSource` here is the `id` of the Rule which is always a number and never a Type. Should this be something like `world.rules[rule.getSource()]` or should we add that data to Rules themselves?